controllers/api/resourceRoutes.js

```diff
@@ -1,25 +1,93 @@
 const router = require('express').Router();
 const { Resource } = require('../../models');
 const { withApprovedMembership, withAuth } = require('../../utils/auth');
+const upload = require('../../config/upload');
+// uploads go to the public/upload directory. Included in gitIgnore
 
 const basePath = '/:space_id/idea/:idea_id/resource';
+// image posting
+router.post(
+  `${basePath}/image`,
```

Comment:
Struggling to find a solution here. 

I have the static values of fieldname and mimetype to work with from req.body. 
![image](https://user-images.githubusercontent.com/92805933/158278323-f5a31441-bb6c-4a6c-b420-5d383ad8f1f0.png)

I can't call on req.body before the route. 
The multer route requires this upload.single('name') before the req.res.

 The name must match the name field from html. 
I can't really use the static 'type' from html, because it's the same for an image as a doc (file). If you change the type to image or anything other than file you lose the file picker. 

So while I can dynamically alter the content and file type on lines 40 and 41, I can't figure out how to alter the name match on line 17. Any suggestions?
![image](https://user-images.githubusercontent.com/92805933/158278256-b3c4de5f-f6fd-4b37-a21c-64fbea868335.png)

Reply:
Issues created below for this per our convo.